test/0.8.9/withdrawal-queue-gas.test.js

```diff
@@ -0,0 +1,192 @@
+/* eslint-disable no-template-curly-in-string */
+const { contract, ethers } = require('hardhat')
+const { bn } = require('@aragon/contract-helpers-test')
+const { itParam } = require('mocha-param')
+
+const { ETH, StETH, shares } = require('../helpers/utils')
+const { setBalance, EvmSnapshot } = require('../helpers/blockchain')
+const { deployWithdrawalQueue } = require('./withdrawal-queue-deploy.test')
+
+contract('WithdrawalQueue', ([owner, user]) => {
+  let wq, steth, defaultShareRate, belowShareRate, aboveShareRate, snapshot
+  let gasPrice = 1
+  const currentRate = async () =>
+    bn(await steth.getTotalPooledEther())
+      .mul(bn(10).pow(bn(27)))
+      .div(await steth.getTotalShares())
+
+  const MAX_BATCH_SIZE = 280
+  const batchIncrement = (i) => i * 2
+  const REQ_AMOUNT = ETH(0.00001)
+  const batchSizes = []
+  for (let batch_size = 1; batch_size <= MAX_BATCH_SIZE; batch_size = batchIncrement(batch_size)) {
+    batchSizes.push(batch_size)
+  }
+
+  before('Deploy', async function () {
+    if (!process.env.REPORT_GAS) {
+      this.skip()
+    }
+    snapshot = new EvmSnapshot(ethers.provider)
+
+    const deployed = await deployWithdrawalQueue({
+      stethOwner: owner,
+      queueAdmin: owner,
+      queuePauser: owner,
+      queueResumer: owner,
+      queueFinalizer: owner,
+    })
+
+    steth = deployed.steth
+    wq = deployed.withdrawalQueue
+
+    await steth.setTotalPooledEther(ETH(600))
+    await setBalance(steth.address, ETH(600))
+    await steth.mintShares(user, shares(300))
+    await steth.approve(wq.address, StETH(300), { from: user })
+    defaultShareRate = await currentRate()
+    belowShareRate = defaultShareRate.divn(2)
+    aboveShareRate = defaultShareRate.muln(2)
+    await snapshot.make()
+  })
+
+  after('clean up', async function () {
+    // only rollback if not skipped
+    if (process.env.REPORT_GAS) {
+      await snapshot.rollback()
+    }
+  })
+
+  context('requestWithdrawal', () => {
+    let results
+
+    before(async () => {
+      results = []
+    })
+
+    after(async () => {
+      console.log('requestWithdrawals')
+      console.table(results)
+    })
+
+    itParam('batch size ${value}', batchSizes, async (batch_size) => {
+      const args = [
+        Array(batch_size).fill(REQ_AMOUNT),
+        user,
+        {
+          from: user,
+          gasPrice: gasPrice++,
```

Comment:
Why do we use `gasPrice` as a simple increment counter?

Reply:
gas fee rise with these many txs, so it's a simple hack